lib/diagnostic.rb

```diff
@@ -5,36 +5,38 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    lengths = []
+    list.each { |element| lengths.push(element.length) }
+    list[lengths.index(lengths.max)]
```

Comment:
This may work, but since the question wanted a comparison, we should use the spaceship operator

`list.max { |a, b| a.length <=> b.length }`